File: #1-25 (C, Java, Python)/Python/Euler22.py
#Euler22 ~ Names scores 
import timeit
start = timeit.default_timer()

nameList = open("names.txt",'r')
names = sorted(nameList.read().replace('"','').split(','),key=str)

points =  {"A":1, \
           "B":2, \
           "C":3, \
           "D":4, \
           "E":5, \
           "F":6, \
           "G":7, \
           "H":8, \
           "I":9, \
           "J":10, \
           "K":11, \
           "L":12, \
           "M":13, \
           "N":14, \
           "O":15, \
           "P":16, \
           "Q":17, \
           "R":18, \
           "S":19, \
           "T":20, \
           "U":21, \
           "V":22, \
           "W":23, \
           "X":24, \
           "Y":25, \
           "Z":26}

sum = 0
for i,word in enumerate(names):
    p = 0
    for l in word:
        p += points[l]
    sum += p*(i+1) #Multiply points by index position & add to sum
    
stop = timeit.default_timer()
print("Solution: ",sum,"\n")
print("Program completed in","{0:.3f}".format(stop - start), "seconds.")

File: #1-25 (C, Java, Python)/Java/Euler22.java
//Euler22 ~ Names scores
import java.util.*;
import java.io.File;
import java.io.FileNotFoundException;

public class Euler22
{
    public static void main(String args[])
	{
		long startTime = System.nanoTime();
		
        try
        {
			Scanner nameList = new Scanner(new File("names.txt")).useDelimiter("[\",]+");
			ArrayList<String> names = new ArrayList<String>();
			while (nameList.hasNext())
			{
				names.add(nameList.next());
			}
			Collections.sort(names); //Alphabetically sort names in the list
            
			int w, x, y;
			int z = 0;
			for(int i = 0; i <= names.size() - 1; i ++)
			{
				x = 0;
				String letters = names.get(i);
				for(int j = 0; j <= letters.length() - 1; j ++)
				{
					w = points(letters.charAt(j));
					x += w;
				}
				y = x * (i + 1);
				z += y;
            }
		   
			long endTime = System.nanoTime();
			long timeElapsed = endTime - startTime;
			System.out.println("Solution: " + z + "\n");
			System.out.println("Program completed in " + timeElapsed/1000000000 + " seconds.");
        }
        catch(FileNotFoundException err)
        {
           System.out.println("File names.txt must be in the same directory");
        }
    }
    
    public static int points(char l)
    {
        String letter = Character.toString(l);
        Map<String, Integer> p = new HashMap<String, Integer>();
        p.put("A", 1);
        p.put("B", 2);
        p.put("C", 3);
        p.put("D", 4);
        p.put("E", 5);
        p.put("F", 6);
        p.put("G", 7);
        p.put("H", 8);
        p.put("I", 9);
        p.put("J", 10);
        p.put("K", 11);
        p.put("L", 12);
        p.put("M", 13);
        p.put("N", 14);
        p.put("O", 15);
        p.put("P", 16);
        p.put("Q", 17);
        p.put("R", 18);
        p.put("S", 19);
        p.put("T", 20);
        p.put("U", 21);
        p.put("V", 22);
        p.put("W", 23);
        p.put("X", 24);
        p.put("Y", 25);
        p.put("Z", 26);
        
        int value = p.get(letter);
        return value;
    }
}